### ISSUE TITLE
Vision Transformer相关的代码问题

### ISSUE THREAD

**Issue by SadMemories** [WZMIAOMIAO/deep-learning-for-image-processing#548]
**System information**
* Have I written custom code: No
* OS Platform(e.g., window10 or Linux Ubuntu 16.04): Ubuntu 18.04
* Python version: 3.7
* Deep learning framework and version(e.g., Tensorflow2.1 or Pytorch1.3): Pytorch 1.11
* Use GPU or not: yes
* CUDA/cuDNN version(if you use GPU): cuda 11.3
* The network you trained(e.g., Resnet34 network): Vision Transformer

**Describe the current behavior**
博主，你好，我在阅读 **pytorch_classification/vision_transformer** 代码的时候遇到了几个问题：
1. 在vit_model.py文件中，初始化使用了nn.init.trunc_normal_，为什么在初始化时要使用截断分布，使用正常的正态分布不可以吗？
2. 在utils.py文件的133行使用了accu_loss += loss.detach()，这里为什么要使用loss.detach()，accu_loss也没有进行反向传播的操作，直接使用accu_loss += loss不行吗？
3. 在原论文中需要对梯度进行裁剪，具体的语句是 **grad clipping at global norm 1**，但是在您写的代码中我并没有看到梯度裁剪的相关语句，请问是没有加吗？

期待您的回答，非常感谢！
**Error info / logs**

**Comment by WZMIAOMIAO** [comment]
1. 该项目代码参考的是timm库，你可以问下原作者：
https://github.com/rwightman/pytorch-image-models/blob/20a1fa63f8ea999dab29d927d5e1866ed3b67348/timm/models/vision_transformer.py#L455-L462

2. 统计loss时不需要梯度信息，所以用detach切断梯度，如果不想用detach也可以用item()方法
3. 我提供的代码中没有对梯度进行裁剪，如果需要可以自行加入。

**Comment by SadMemories** [comment]
感谢回答

**Comment by xiaobaitubaibaibai** [comment]
![image](https://user-images.githubusercontent.com/90377527/168509310-d40a3b51-6e86-4631-a927-5c9406f3e2b8.png)
![image](https://user-images.githubusercontent.com/90377527/168509404-fad75c37-414d-4597-990e-7c31d1e16fa2.png)
Dear author:I want to use LambdaLR,the first picture is my changed code,the second picture is error message.I would appreciate it if you would help me.Thank you
